### DISCUSSION TITLE
Problems whit len

### DISCUSSION THREAD

**Discussion by be80be** [micropython/micropython#15561] [General]
Here the code go easy on me I've not used micrpyhton but little over week.

``
import machine
from ssd1306 import SSD1306_I2C
import framebuf

i2c = machine.I2C(scl=machine.Pin(5), sda=machine.Pin(4))

oled = SSD1306_I2C(128, 64, i2c)# ... (I2C initialization and OLED object creation as before)

def draw_char(char, x, y, size):

    char_data = {
        'A': [0b00100, 0b01110, 0b11111, 0b11111, 0b01110],
  
    }

    char_bitmap = char_data.get(char, None)
    if char_bitmap:
        for row in range(len(char_bitmap)):
            for col in range(len(char_bitmap[0])):
                oled.pixel(x + col * size, y + row * size, char_bitmap[row] & (1 << col))

oled.fill(0)
draw_char('A', 20, 10, 3)  # Draw 'A' with size 3
oled.show()``

And i get these errors 
MPY: soft reboot
Warning: I2C(-1, ...) is deprecated, use SoftI2C(...) instead
Traceback (most recent call last):
  File "<stdin>", line 26, in <module>
  File "<stdin>", line 21, in draw_char
TypeError: object of type 'int' has no len()

**Comment by bixb922** [comment]
Please format with 3 backticks.

```char_bitmap[0]``` is ```0b00100``` which is a ```int```. There is no ```len()``` for a ```int```.

**Reply by be80be** [reply]
Thanks  it works now next problem is making fonts into a bitmap

**Reply by robert-hh** [reply]
See https://github.com/peterhinch/micropython-font-to-py
